drake/solvers/Optimization.h

```diff
@@ -0,0 +1,323 @@
+#ifndef DRAKE_OPTIMIZATION_H
+#define DRAKE_OPTIMIZATION_H
+
+#include "Core.h"
+#include <list>
+#include <memory>
+#include <initializer_list>
+
+namespace Drake {
+
+  struct DecisionVariable {
+    enum VarType {
+      CONTINUOUS,
+      INTEGER,
+      BINARY
+    };
+    VarType type;
+    std::string name;
+    Eigen::VectorXd value;
+    size_t start_index;
```

Comment:
Document this? I assume `start_index` is the index of DecisionVariable.value(0) in the aggregate decision variables.